src/libs/actions/Report.js

```diff
@@ -442,6 +442,10 @@ function fetchOrCreateChatReport(participants) {
     })
 
         .then((data) => {
+            if (data.jsonCode !== 200) {
+                throw new Error(data.message);
+            }
```

Comment:
So if I alert here and do a return it will fall into the next `.then` and then i'd have to handle it there as well hence duplicating some code. By doing a throw and a catch it keeps the flow clean. 

> Maybe something else causes an error here and you are now alerting whatever it is in the catch()

Not sure why this is a bad thing. If we see an interesting alert of a case we don't want to handle or catch differently then we update the flow accordingly when that happens.

Reply:
>it will fall into the next .then 

Not too sure what you mean as there is no "next `.then()`"?

>Not sure why this is a bad thing

It's bad because you are alerting to the user a message that is not customer facing and could be ... anything?

Reply:
> Not too sure what you mean as there is no "next .then()"?

So if i replace,
```
throw new Error(data.message);
```
with,
```
alert(data.message);
return;
```
then it will fall into this ["then"](https://github.com/Expensify/Expensify.cash/blob/e05454f16a2d8d01f9b513bfa81457739de225f5/src/libs/actions/Report.js#L461-L481) block and throw an error because `data.reports[reportID]` does not exist. So then I've to have an early return there too when data is not what we expect. Hence doing a throw from the first "then" block and then using a catch felt clean. 

> It's bad because you are alerting to the user a message that is not customer facing and could be ... anything?

Well i guess that's the issue with this first phase solution. Either i throw an alert for any flow or handle the error redundantly in those [two "then" conditions](https://github.com/Expensify/Expensify.cash/blob/e05454f16a2d8d01f9b513bfa81457739de225f5/src/libs/actions/Report.js#L444-L481). So yeah let me know your thoughts. One solution i can think off is to save a list of accepted alert messages in CONST then have the catch error check if its part of those to display it. If its not then it would do a console error log instead. How does that sound?

Reply:
I like the idea about checking for a specific constant error message. But I think it would be easier to just alert when you do not have the necessary result. If there is no `data` in the second `.then()` then return early again and that will solve the problem.

Reply:
Alright, updated this to alert when the first "then" result is not what i expect and I've updated the next "then" function to return early when there are no reports.